src/firefly/js/tables/DecimateInfo.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+
+const DECIMATE_TAG='decimate';
+
+/**
+ * Created by tatianag on 3/14/16.
+ * This is a helper class to deal with decimation parameters.
+ * Decimation allows to reduce the number of the points to be plotted.
+ */
+export class DecimateInfo {
+    constructor(xColumnName, yColumnName, maxPoints, xyRatio, xMin, xMax, yMin, yMax) {
+        this.xColumnName = xColumnName;
+        this.yColumnName = yColumnName;
+        this.maxPoints = Number.parseInt(maxPoints,10); //NaN if not defined
+        this.xyRatio = Number.parseFloat(xyRatio);
+        this.xMin = Number.parseFloat(xMin);
+        this.xMax = Number.parseFloat(xMax);
+        this.yMin = Number.parseFloat(yMin);
+        this.yMax = Number.parseFloat(yMax);
+    }
+
+    static isValid(decimateInfo) {
+        return decimateInfo && this.xColumnName && this.yColumnName;
+    }
+
+    static parse(str) {
+        if (!str) return;
+
+        const kv = str.split('=', 2);
+        let dinfo = null;
+        if (kv && kv.length === 2 && kv[0].equals(DECIMATE_TAG)) {
+            dinfo = new DecimateInfo(...kv[1].split(','));
+            return dinfo.isValid() ? dinfo : null;
```

Comment:
Good catch. I forgot to test the converted code and since we don't use parse on the client, I got away with it.